room.h

```diff
@@ -0,0 +1,37 @@
+#ifndef ROOM_H
+#define ROOM_H
+
+#include <unordered_map>
+#include "sensor.h"
+#include "device.h"
+
+class Room
+{
+public:
+    Room(const std::string &name = "");
+
+    void AddSensor(Sensor &sensor);
+    void DeleteSensor(const std::string &name);
+    Sensor* GetSensor(const std::string &name);
+
+    void AddDevice(Device &device);
+    void DeleteDevice(const std::string &name);
+    Device* GetDevice(const std::string &name);
+
+    std::string GetName();
+    void SetName(const std::string &name);
+
+    std::unordered_map<std::string, Sensor> GetSensors();
```

Comment:
Have a look at benefits of using ref/const ref returned instead of returning the value.

Reply:
I will